### ISSUE TITLE
[htdf2.0] query Tx and Block  response should included tendermint Events

### ISSUE THREAD

**Issue by youngqqcn** [orientwalt/htdf#71]
Tx query response should included tendermint Events details to replace htdf1.x Tags . 
Especially in delegateion related transctions, such as  `Delegate` , `Undelegate` , `Withdraw Delegate Rewards` .

**Comment by youngqqcn** [comment]
Block query response also should contains Events for  `Undelegate Complete`

**Comment by youngqqcn** [comment]
Block Events:

http://127.0.0.1:1317/block_detail/921

```json
{
    "block_meta": {
        "block_id": {
            "hash": "A5175A9F03EB1B26F3F03EA5C23F909222582502C913AF84928E96D45CD7F9CF",
            "parts": {
                "total": "1",
                "hash": "1093D5E62B32ADE4B7D7C9D6785262E46F1CAB833A81F80ABC4011B016F5690B"
            }
        },
        "block_size": "485",
        "header": {
            "version": {
                "block": "10",
                "app": "0"
            },
            "chain_id": "testchain",
            "height": "921",
            "time": "2021-05-12T07:22:27.109357181Z",
            "last_block_id": {
                "hash": "CAE7AFE3C6898B0D359BA5D184784A59A52ECF4A9032976EDFEB0AE6B3D38285",
                "parts": {
                    "total": "1",
                    "hash": "5DC99A223BA645BE6230FA1CDDF9754C38F29623DF73596B2E26D7B545103A13"
                }
            },
            "last_commit_hash": "C4182ACC35B2856287155CCCB1C929F934D063CCD42957E4265E8764489B2D9B",
            "data_hash": "",
            "validators_hash": "673B0984A793347DD7E845A43F1F3DF134BE3CA203BEFACCE03B492AD1666C51",
            "next_validators_hash": "673B0984A793347DD7E845A43F1F3DF134BE3CA203BEFACCE03B492AD1666C51",
            "consensus_hash": "4A3999F2F89E2E3AB1198825F730FCAE48D21406F7B9E36813A76B9A9B5ED509",
            "app_hash": "F77D5B241CE21908D1527CE3C6F8EE09A20A8DCC5B3B5EB529038295828A2D17",
            "last_results_hash": "",
            "evidence_hash": "",
            "proposer_address": "CB9DA77521A2C96D19612DB5188FB9DCA7D25A5D"
        },
        "num_txs": "0"
    },
    "block": {
        "txs": null,
        "evidence": {
            "evidence": null
        },
        "last_commit": {
            "height": "920",
            "round": "0",
            "block_id": {
                "hash": "CAE7AFE3C6898B0D359BA5D184784A59A52ECF4A9032976EDFEB0AE6B3D38285",
                "parts": {
                    "total": "1",
                    "hash": "5DC99A223BA645BE6230FA1CDDF9754C38F29623DF73596B2E26D7B545103A13"
                }
            },
            "signatures": [
                {
                    "block_id_flag": 2,
                    "validator_address": "CB9DA77521A2C96D19612DB5188FB9DCA7D25A5D",
                    "timestamp": "2021-05-12T07:22:27.109357181Z",
                    "signature": "NuX/5dybb2Sy6p6xr0l1oELHIvtpT4PBD5xqcakfmKEwoQAMJR3/aqVRgkLKNibOuk2TPne9XKJ1HDJKoBesCA=="
                }
            ]
        }
    },
    "events": [
        {
            "type": "complete_unbonding",
            "attributes": [
                {
                    "key": "amount",
                    "value": "200000000000satoshi"
                },
                {
                    "key": "validator",
                    "value": "htdfvaloper1gu23408yyv6lk6vqjkykecmulnj0xsmhqr47hs"
                },
                {
                    "key": "delegator",
                    "value": "htdf1xwpsq6yqx0zy6grygy7s395e2646wggufqndml"
                }
            ]
        }
    ],
    "time": "2021-05-12 15:22:27"
}

```

**Comment by youngqqcn** [comment]
Transaction Events:

http://127.0.0.1:1317/txs/4A0185ADE89739348B79AC3DB84FF90127DF87F552F9DABFED2651AED93793E1

```json

{
    "height": "897",
    "txhash": "4A0185ADE89739348B79AC3DB84FF90127DF87F552F9DABFED2651AED93793E1",
    "data": "0C08B284EE840610C2B7A5E501",
    "raw_log": "[{\"msg_index\":\"0\",\"success\":true,\"log\":\"\"}]",
    "logs": [
        {
            "msg_index": "0",
            "success": true,
            "log": ""
        }
    ],
    "gas_wanted": "30000",
    "gas_used": "30000",
    "events": [
        {
            "type": "message",
            "attributes": [
                {
                    "key": "action",
                    "value": "begin_unbonding"
                },
                {
                    "key": "module",
                    "value": "staking"
                },
                {
                    "key": "sender",
                    "value": "htdf1xwpsq6yqx0zy6grygy7s395e2646wggufqndml"
                }
            ]
        },
        {
            "type": "unbond",
            "attributes": [
                {
                    "key": "delegator",
                    "value": "htdf1xwpsq6yqx0zy6grygy7s395e2646wggufqndml"
                },
                {
                    "key": "validator",
                    "value": "htdfvaloper1gu23408yyv6lk6vqjkykecmulnj0xsmhqr47hs"
                },
                {
                    "key": "amount",
                    "value": "200000000000"
                },
                {
                    "key": "completion_time",
                    "value": "2021-05-12T07:22:26Z"
                }
            ]
        },
        {
            "type": "withdraw_rewards",
            "attributes": [
                {
                    "key": "module",
                    "value": "distr"
                },
                {
                    "key": "delegator",
                    "value": "htdf1xwpsq6yqx0zy6grygy7s395e2646wggufqndml"
                },
                {
                    "key": "validator",
                    "value": "htdfvaloper1gu23408yyv6lk6vqjkykecmulnj0xsmhqr47hs"
                },
                {
                    "key": "withdraw_address",
                    "value": "htdf1xwpsq6yqx0zy6grygy7s395e2646wggufqndml"
                },
                {
                    "key": "amount",
                    "value": "353160029"
                }
            ]
        }
    ],
    "tx": {
        "type": "auth/StdTx",
        "value": {
            "msg": [
                {
                    "type": "htdf/MsgUndelegate",
                    "value": {
                        "delegator_address": "htdf1xwpsq6yqx0zy6grygy7s395e2646wggufqndml",
                        "validator_address": "htdfvaloper1gu23408yyv6lk6vqjkykecmulnj0xsmhqr47hs",
                        "amount": {
                            "denom": "satoshi",
                            "amount": "200000000000"
                        }
                    }
                }
            ],
            "fee": {
                "gas_wanted": "30000",
                "gas_price": "100"
            },
            "signatures": [
                {
                    "pub_key": {
                        "type": "tendermint/PubKeySecp256k1",
                        "value": "A/tHuncwUGZHLxKdevqasdS+DCcW8fbt0Sv8sRpezqWy"
                    },
                    "signature": "eJb6HH6txlH5yBpFHiaZ/kjy/q+vbP6PLuiS/dgC5ShioDOj+THMm+PZ08Uv7Uiigm5W0XYl2OOIRq9QYtz7yg=="
                }
            ],
            "memo": ""
        }
    },
    "timestamp": "2021-05-12T07:20:26Z"
}
```

**Comment by youngqqcn** [comment]
fix at commit 9291bedaa856759f9efe0af770b2d2175bd3e57e

**Comment by youngqqcn** [comment]
if no delegations ( or all delegation in `unbonding` status) the response as below

```
curl localhost:1317/distribution/delegators/htdf1g0cmfdja0c08jkchmaqq598uc7jmhw0e30pm7q/rewards/htdfvaloper1jwgndhkts6ez49p6dd7egye4mhmtslqt3rhnyc

{"error":"{\"codespace\":\"sdk\",\"code\":1,\"message\":\"delegation does not exist\"}"}
```

```
curl localhost:1317/staking/delegators/htdf1g0cmfdja0c08jkchmaqq598uc7jmhw0e30pm7q/delegations
null
```
